federation/pkg/federation-controller/cluster/clustercontroller.go

```diff
@@ -184,6 +187,14 @@ func (cc *ClusterController) UpdateClusterStatus() error {
 				}
 			}
 		}
+		clusterClientSet, _ := NewClusterClientset(&cluster)
+		zones, region, err := GetClusterZones(clusterClientSet)
+		if err != nil {
+			glog.Errorf("Failed to get zones and region for cluster %s: %v", cluster.Name, err)
```

Comment:
At line 193, I intentionally did not return a error, as the rest of the Status update should continue, and these zone labels will be picked up on a subsequent iteration, if they exist.  I'll add a comment, and perhaps demote this log message to a warning.  For line 202, you're right, we should return an error.  That was existing code.  But I can fix it here I guess.